tests/tests_geomstats/test_pullback_metric.py

```diff
@@ -0,0 +1,307 @@
+"""Unit tests for the pull-back metrics."""
+
+import warnings
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.geometry.pullback_metric import PullbackMetric
+
+
+class TestPullbackMetric(geomstats.tests.TestCase):
+    def setUp(self):
+        warnings.simplefilter('ignore', category=UserWarning)
+        gs.random.seed(0)
+        self.dim = 2
+        self.sphere = Hypersphere(dim=self.dim)
+        self.sphere_metric = self.sphere.metric
+
+        def _sphere_immersion(spherical_coords):
```

Comment:
Why didn't you use `Hypersphere.spherical_to_extrinsic` directly?

Reply:
To avoid relying on other part of the code, i.e. to really have a unit-test.